docs/raw/9-blockchain-interactions.md

```diff
@@ -0,0 +1,186 @@
+
+
+```
+func (ec *Client) SendTransaction(ctx context.Context, tx *types.Transaction) error 
+```
+
+https://github.com/ethereum/go-ethereum/blob/26d271dfbba1367326dec38068f9df828d462c61/ethclient/ethclient.go#L512
+
+### Fetching balance
+
+#### BlockByHash
+
+`BlockByHash` returns the given full block.
+
+It is used by status to fetch a given block which will then be inspected for 
+transfers to the user address, both tokens and ETH.
+
+```
+func (ec *Client) BlockByHash(ctx context.Context, hash common.Hash) (*types.Block, error)
+```
+
+https://github.com/ethereum/go-ethereum/blob/26d271dfbba1367326dec38068f9df828d462c61/ethclient/ethclient.go#L78
+
+#### BlockByNumber
+
+`BlockByNumber` returns a block from the current canonical chain. If number is nil, the
+latest known block is returned.
+
+```
+func (ec *Client) BlockByNumber(ctx context.Context, number *big.Int) (*types.Block, error)
+```
+
+https://github.com/ethereum/go-ethereum/blob/26d271dfbba1367326dec38068f9df828d462c61/ethclient/ethclient.go#L82
+
+#### FilterLogs
+
+`FilterLogs` executes a filter query.
+
+Status uses this function to filter out logs, using the hash of the block
+and the address that we are interested in, both inbound and outbound.
+
+```
+func (ec *Client) FilterLogs(ctx context.Context, q ethereum.FilterQuery) ([]types.Log, error) 
+```
+
+https://github.com/ethereum/go-ethereum/blob/26d271dfbba1367326dec38068f9df828d462c61/ethclient/ethclient.go#L377
+
+
+#### NonceAt
+
+`NonceAt` returns the account nonce of the given account.
+
+```
+func (ec *Client) NonceAt(ctx context.Context, account common.Address, blockNumber *big.Int) (uint64, error)
+```
+
+https://github.com/ethereum/go-ethereum/blob/26d271dfbba1367326dec38068f9df828d462c61/ethclient/ethclient.go#L366
+
+
+#### TransactionByHash
+
+`TransactionByHash` returns the transaction with the given hash, used to inspect those 
+transaction made/received by the user.
+
+```
+func (ec *Client) TransactionByHash(ctx context.Context, hash common.Hash) (tx *types.Transaction, isPending bool, err error)
+```
+
+https://github.com/ethereum/go-ethereum/blob/26d271dfbba1367326dec38068f9df828d462c61/ethclient/ethclient.go#L202
+
+#### HeaderByNumber
+
+`HeaderByNumber` returns a block header from the current canonical chain.
+
+```
+func (ec *Client) HeaderByNumber(ctx context.Context, number *big.Int) (*types.Header, error) 
+```
+
+https://github.com/ethereum/go-ethereum/blob/26d271dfbba1367326dec38068f9df828d462c61/ethclient/ethclient.go#L172
+
+
+#### TransactionReceipt
+
+`TransactionReceipt` returns the receipt of a transaction by transaction hash.
+It is used in status to check if a token transfer was made to the user address.
+
+```
+func (ec *Client) TransactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error)
+```
+
+https://github.com/ethereum/go-ethereum/blob/26d271dfbba1367326dec38068f9df828d462c61/ethclient/ethclient.go#L270
+
+
+## ENS
```

Comment:
Elaborate on purpose of ENS and how we use it, how can I have a Status name and get it to show up in my client, etc